### PR TITLE
Make dev env LAN access easier

### PR DESCRIPTION
**Pull request by merapi**
## Summary

As in title.

## Testing Instructions

---

Fixes #

### CODE REVIEW

**Comment by spacedmonkey** [comment]
This looks good. I had to do the same when testing in windows VM when debugging paste upload. 

However, when testing this, it didn't work. When access the site locally, it just redirects to localhost. See 
```
curl -I http://192.168.86.193:8899/
HTTP/1.1 302 Found
Date: Thu, 02 Jul 2020 09:47:38 GMT
Server: Apache/2.4.38 (Debian)
X-Powered-By: PHP/7.4.7
Location: http://localhost:8899/
Content-Type: text/html; charset=UTF-8
```

Did this work locally for you @merapi ?

I think there is a simple workaround, that I have used before. 
Add something like [this](https://github.com/spacedmonkey/wp-base-docker/blob/f37cc784bc77a04bc620178b1249b7d2b1a18a80/wp-config.php#L158-L181) in the wp-config.php file. 

What this does it dynamically set the home/ site urls to what host headers are passes. This means that site will not redirect to whatever the localhost.

File: bin/local-env/docker-compose.yml [hunk 1]
```diff
@@ -4,7 +4,7 @@ services:
   wordpress:
     image: wordpress
     ports:
-      - '127.0.0.1:8899:80'
+      - '8899:80'
```
Comment (spacedmonkey) [inline bin/local-env/docker-compose.yml:7]: @swissspidy Can you explain why this IP was put here in the first place?

Comment (swissspidy) [reply bin/local-env/docker-compose.yml:7]: Basically to prevent exactly what this PR does (access from other devices in the network) :-)

**Comment by merapi** [comment]
@spacedmonkey This works for me, browser/os cache can be involved in this, but on a clean slate - never had a problem (just had to go to localhost and change the URLs to my IP, and then access the new site). I was only redirected to localhost if I didn't change WP General Settings URL - did you?

**Comment by spacedmonkey** [comment]
> @spacedmonkey This works for me, browser/os cache can be involved in this, but on a clean slate - never had a problem (just had to go to localhost and change the URLs to my IP, and then access the new site). I was only redirected to localhost if I didn't change WP General Settings URL - did you?

I think it would be better, if we can the home / site url dynamic so you don't have to do this.

**Comment by merapi** [comment]
> > @spacedmonkey This works for me, browser/os cache can be involved in this, but on a clean slate - never had a problem (just had to go to localhost and change the URLs to my IP, and then access the new site). I was only redirected to localhost if I didn't change WP General Settings URL - did you?
> 
> I think it would be better, if we can the home / site url dynamic so you don't have to do this.

Without a doubt. If there is an easy and reliable way to do this - 🙏 . 
My concern was about getting the right local IP (which interface?) on all OSs - in that solution we don't need to care about it?

**Comment by spacedmonkey** [comment]
> My concern was about getting the right local IP (which interface?) on all OSs - in that solution we don't need to care about it?

It is not an OS issue. The host header is handled via apache in the docker. So apache is the only place that it needs to detect host header correctly is there.

File: bin/local-env/install-wordpress.sh [hunk 2]
```diff
@@ -7,6 +7,7 @@ set -e
 WP_DEBUG=${WP_DEBUG-true}
 SCRIPT_DEBUG=${SCRIPT_DEBUG-true}
 WP_VERSION=${WP_VERSION-"latest"}
+HOST=${HOST-"localhost"}
```
Comment (spacedmonkey) [inline bin/local-env/install-wordpress.sh]: @swissspidy What you think about this? Make it easy to change the site url. Am I correct in thinking we can use env vars here?

Comment (swissspidy) [reply bin/local-env/install-wordpress.sh]: Yes, those are env vars.  `HOST` (vs. `WP_HOST`) might be a bit ambiguous, but if it works, it works 👍

Comment (spacedmonkey) [reply bin/local-env/install-wordpress.sh]: Fixed.

**Review by spacedmonkey — APPROVED** [review]
👍

**Comment by merapi** [comment]
From my testing, it doesn't work as expected.

Started the env and accessed it from my phone via http://192.168.43.11:8899/ (this works because of my docker-compose.yml change) now `Log in` link redirects to localhost:8899 (we have hardcoded `WP_HOST=${WP_HOST-"localhost"}`) and because there is no localhost on my phone - it doesn't work.

So are we back to
> My concern was about getting the right local IP (which interface?) on all OSs

?

We need to get that IP and feed it into `WP_HOST=${WP_HOST-"192.168.43.11"}` or do the trick with `$_SERVER['HTTP_HOST']`, right? @spacedmonkey

**Pull request closed without merging**